core/src/main/java/org/infinispan/interceptors/distribution/ScatteredDistributionInterceptor.java

```diff
@@ -714,7 +712,7 @@ private Object handlePutMapOnOrigin(InvocationContext ctx, PutMapCommand command
             localEntries.put(key, new MetadataImmortalCacheValue(entry.getValue(), metadata));
             commitSingleEntryIfNewer(ctxEntry, ctx, command);
          } else if (info.primary() == null) {
-            throw AllOwnersLostException.INSTANCE;
+            throw new AllOwnersLostException(cacheTopology.getTopologyId());
```

Comment:
The way I see it, a key is lost iif `cache.get(key)` returns `null` or throws an `AvailabilityException`. Scattered cache may have a single owner in the CH, but it behaves as if it had 1 primary owner and 1 backup owner.